File: py/soal2.py
# ANGGOTA KELOMPOK
# - Naufal Fahrezi (14081020035)
# - Amalia Nur Fitri (140810200045)
# - Calvin Calfi Montolalu (140810200053)
# - Deani Puteri Virdiana (140810200059)

class Matriks :
    def __init__(self, b, k) :
        self.baris=b
        self.kolom=k
        self.matriks=[b][k]

    def setBaris(self,b) : self.baris=b
    def setKolom(self,k) : self.kolom=k

    def getBaris(self) : return self.baris
    def getKolom(self) : return self.kolom
    
    def input(self) :
        temp_row=[]
        A = []
        self.baris= int(input("Masukkan jumlah baris: "))
        self.kolom= int(input("Masukkan jumlah kolom: "))
        for i in range(0, self.baris) :
            for j in range(0, self.kolom) :
                elemen=input("Masukkan baris ke-" + str(i+1) + " kolom ke-" + str(j+1) + " : ")
                temp_row.append(elemen)
            A.append(temp_row)
            temp_row=[]
        self.matriks = A

    def print(self) :
        for i in range(0, self.baris) :
            for j in range(0, self.kolom) : 
                print(self.matriks[i][j], end="\t")
            print()

    def getNilaiBaris(self) :
        nilaiB=[]
        for i in range(0, self.baris) :
            temp=0
            for j in range(0, self.kolom) :
                temp += int(self.matriks[i][j])
            nilaiB.append(temp)
        
        for i in range(0, self.baris) :
            print(nilaiB[i])


    def getNilaiKolom(self) :
        nilaiK=[]
        for i in range(0, self.baris) :
            temp=0
            for j in range(0, self.kolom) :
                temp += int(self.matriks[j][i])
            nilaiK.append(temp)
        
        for i in range(0, self.kolom) :
            print(nilaiK[i], end="\t")

A = Matriks(0,0)
A.input()
print("\nMatriks A")
A.print()

print("\nJumlah Nilai Baris Matriks A")
A.getNilaiBaris()
print("\nJumlah Nilai Kolom Matriks A")
A.getNilaiKolom()

File: cpp/soal2.cpp
#include <iostream>
#include <vector>

using namespace std;

class Matriks {
    private:
        int baris;
        int kolom;
        int nilai[][100];

    public:
        Matriks() {
            this->baris=0;
            this->kolom=0;
        }

        Matriks(int b, int k) {
            this->baris=b;
            this->kolom=k;
        }

        int getBaris() { return this->baris; }
        int getKolom() { return this->kolom; }

        void input() {
            cout << "Masukkan jumlah baris: "; cin >> this->baris;
            cout << "Masukkan jumlah kolom: "; cin >> this->kolom;
            for(int i=0; i<this->baris; i++) {
                for(int j=0; j<this->kolom; j++) {
                    cout << "Masukkan baris-" << (i+1) << " kolom-" << (j+1) << " : "; cin >> this->nilai[i][j];
                }
            }
        }

        void print() {
            for(int i=0; i<this->baris; i++) {
                for(int j=0; j<this->kolom; j++) {
                    cout << this->nilai[i][j] << "\t";
                }
                cout << endl;
            }
        }

        vector<int> getNilaiBaris() {
            vector<int> nilaiB;
            for(int i=0; i<this->baris; i++) {
                int temp=0;
                for(int j=0; j<this->kolom; j++) {
                    temp += this->nilai[i][j];
                }
                nilaiB.push_back(temp);
            }
            return nilaiB;
        }

        vector<int> getNilaiKolom() {
            vector<int> nilaiK;
            for(int i=0; i<this->baris; i++) {
                int temp=0;
                for(int j=0; j<this->kolom; j++) {
                    temp += this->nilai[j][i];
                }
                nilaiK.push_back(temp);
            }
            return nilaiK;
        }
};

int main() {
    Matriks A;

    A.input();
    cout << "\nMatriks A\n";
    A.print();
    vector<int> jumlahBaris=A.getNilaiBaris();
    vector<int> jumlahKolom=A.getNilaiKolom();
    cout << "\nJumlah baris matriks A :\n";
    for(int baris: jumlahBaris) {
        cout << baris << "\n";
    }

    cout << "\nJumlah kolom matriks A :\n";
    for(int kolom: jumlahKolom) {
        cout << kolom << "\t";
    }
}